alt_installer.py

```diff
@@ -0,0 +1,756 @@
+        "Ensuring check system dependencies choice…",
+        app=app
+    )
+    logging.debug('- config.SKIP_DEPENDENCIES')
+
+    logging.debug(f"> {config.SKIP_DEPENDENCIES=}")
+
+
+def ensure_installation_config(app=None):
+    config.INSTALL_STEPS_COUNT += 1
+    ensure_check_sys_deps_choice(app=app)
+    config.INSTALL_STEP += 1
+    update_install_feedback("Ensuring installation config is set…", app=app)
+    logging.debug('- config.LOGOS_ICON_URL')
+    logging.debug('- config.LOGOS_ICON_FILENAME')
+    logging.debug('- config.LOGOS_VERSION')
+    logging.debug('- config.LOGOS64_MSI')
+    logging.debug('- config.LOGOS64_URL')
+
+    # Set icon variables.
+    app_dir = Path(__file__).parent
+    logos_icon_url = app_dir / 'img' / f"{config.FLPRODUCTi}-128-icon.png"
+    config.LOGOS_ICON_URL = str(logos_icon_url)
+    config.LOGOS_ICON_FILENAME = logos_icon_url.name
+    config.LOGOS64_URL = f"https://downloads.logoscdn.com/LBS{config.TARGETVERSION}{config.VERBUM_PATH}Installer/{config.LOGOS_RELEASE_VERSION}/{config.FLPRODUCT}-x64.msi"  # noqa: E501
+
+    config.LOGOS_VERSION = config.LOGOS_RELEASE_VERSION
+    config.LOGOS64_MSI = Path(config.LOGOS64_URL).name
+
+    logging.debug(f"> {config.LOGOS_ICON_URL=}")
+    logging.debug(f"> {config.LOGOS_ICON_FILENAME=}")
+    logging.debug(f"> {config.LOGOS_VERSION=}")
+    logging.debug(f"> {config.LOGOS64_MSI=}")
+    logging.debug(f"> {config.LOGOS64_URL=}")
+
+    if config.DIALOG == 'tk' and app:
+        send_gui_task(app, 'INSTALL')
+
+
+def ensure_install_dirs(app=None):
+    config.INSTALL_STEPS_COUNT += 1
+    ensure_installation_config(app=app)
+    config.INSTALL_STEP += 1
+    update_install_feedback("Ensuring installation directories…", app=app)
+    logging.debug('- config.INSTALLDIR')
+    logging.debug('- config.WINEPREFIX')
+    logging.debug('- data/bin')
+    logging.debug('- data/wine64_bottle')
+
+    if config.INSTALLDIR is None:
+        config.INSTALLDIR = f"{os.getenv('HOME')}/{config.FLPRODUCT}Bible{config.TARGETVERSION}"  # noqa: E501
+    logging.debug(f"> {config.INSTALLDIR=}")
+    if config.WINEPREFIX is None:
+        config.WINEPREFIX = f"{config.INSTALLDIR}/data/wine64_bottle"
+    logging.debug(f"> {config.WINEPREFIX=}")
+
+    bin_dir = Path(f"{config.INSTALLDIR}/data/bin")
+    bin_dir.mkdir(parents=True, exist_ok=True)
+    logging.debug(f"> {bin_dir} exists: {bin_dir.is_dir()}")
+
+    wine_dir = Path(f"{config.WINEPREFIX}")
+    wine_dir.mkdir(parents=True, exist_ok=True)
+    logging.debug(f"> {wine_dir} exists: {wine_dir.is_dir()}")
+
+    if config.DIALOG == 'tk' and app:
+        send_gui_task(app, 'INSTALLING')
+
+
+def ensure_sys_deps(app=None):
+    config.INSTALL_STEPS_COUNT += 1
+    ensure_install_dirs(app=app)
+    config.INSTALL_STEP += 1
+    update_install_feedback("Ensuring system dependencies are met…", app=app)
+
+    if not config.SKIP_DEPENDENCIES:
+        utils.get_package_manager()
+        utils.check_dependencies()
```

Comment:
Having to repeat this step each time on a cancelled install is something that would be great to skip.

Reply:
Do you mean that there should be some way to record that dependencies have already been checked and so shouldn't need to be checked again later? Otherwise, my normal strategy is to use the `-K` option.

Reply:
I think it would be good to have something in addition to `-K`, yes. Happy to be dissuaded.